<launch>
  <arg name="launch_websocket" default="true" />
  <arg name="debug" default="false" />

  <param name="use_sim_time" value="false" />

  <!-- MongoDB -->
  <param name="mongodb_use_daemon" value="true" />
  <param name="mongodb_host" value="localhost" />
  <param name="mongodb_port" value="27017" />
  <node name="message_store" pkg="mongodb_store" type="message_store_node.py" output="screen" required="true" /> 

  <!-- Web -->
  <include if="$(arg launch_websocket)" file="$(find rosbridge_server)/launch/rosbridge_websocket.launch" />
  <!--node pkg="ros_web_video" type="ros_web_video" name="pbi_video_server" required="true">
    <param name="port" value="8888" />
    <param name="framerate" value="15" />
    <param name="bitrate" value="250000" />
    <param name="profile" value="best" />
  </node>
  <node pkg="tf2_web_republisher" type="tf2_web_republisher" name="tf2_web_republisher" /-->

  <!-- Point cloud -->
  <!--node pkg="nodelet" type="nodelet" name="pbi_convert_metric" args="standalone depth_image_proc/convert_metric">
    <remap from="image_raw" to="pbi_annotator/image_depth" />
    <remap from="image" to="pbi_annotator/image_depth_float" />
  </node-->
  <!--node pkg="depthcloud_encoder" type="depthcloud_encoder_node" name="pbi_depthcloud_encoder">
    <param name="rgb" value="/pbi_annotator/image_color" />
    <param name="depth" value="/pbi_annotator/image_depth" />
    <param name="camera_frame_id" value="/head_mount_kinect_rgb_optical_frame" />
  </node-->
  <!-- Hard-coded camera pose -->
  <node pkg="tf" type="static_transform_publisher" name="pbi_camera_pose_publisher" args="0.022 -0.017 1.747 0.614 -0.613 0.351 -0.352 base_link head_mount_kinect_rgb_optical_frame 10" required="true" />

  <!-- Object tracker -->
  <rosparam command="load" file="$(find task_perception)/config/particle_tracker.yaml" />
  <rosparam command="load" file="$(find task_perception)/config/camera.yaml" />
  <rosparam command="load" file="$(find task_perception)/config/object.yaml" />
  <node pkg="dbot_ros" type="multi_object_tracker_node" name="multi_object_tracker_pringles" args="pringles" required="true" />
  <node pkg="dbot_ros" type="multi_object_tracker_node" name="multi_object_tracker_cheezit" args="cheezit" required="true" />
  <node pkg="dbot_ros" type="initialize_object_action_node" name="initialize_object_action_server" required="true" />

  <!-- Skeleton tracker -->
  <node pkg="nerf_b" type="nerf_tracker_node" name="pbi_skeleton_tracker" required="true" />

  <!-- Annotator -->
  <rosparam command="load" file="$(find task_perception)/config/contact_detection.yaml" />
  <rosparam command="load" file="$(find task_perception)/config/grasp_planner.yaml" />
  <node if="$(arg debug)" pkg="task_perception" type="annotator" name="pbi_annotator" required="true" launch-prefix="xterm -e gdb --args" />
  <node unless="$(arg debug)" pkg="task_perception" type="annotator" name="pbi_annotator" required="true" />
</launch>
